File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int N = nums.length / 2;
        Set<Integer> set = new HashSet<Integer>();
        
        for(int i=0; i < nums.length; i++){
            if (!set.contains(nums[i])) 
                set.add(nums[i]);
            
        }
        int answer = set.size() < N ? set.size() : N;
        
        return answer;
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    N = nums.length / 2
    sett = new Set(nums)
    num_sett = sett.size
    
    if (num_sett < N ){
        answer = num_sett
    } else {
        answer = N
    }
   
       
    return answer;
}